maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -1184,6 +1208,39 @@ private SurefireProperties setupProperties() {
         return result;
     }
 
+    private SurefireProperties calculateEffectiveProperties(
+            Properties systemProperties,
+            Map<String, String> systemPropertyVariables,
+            Properties userProperties,
+            SurefireProperties sysPropsFromFile) {
+        SurefireProperties result = new SurefireProperties();
+        result.copyPropertiesFrom(systemProperties);
+
+        Collection<String> overwrittenProperties = result.copyPropertiesFrom(sysPropsFromFile);
+        if (!overwrittenProperties.isEmpty() && getConsoleLogger().isDebugEnabled()) {
+            getConsoleLogger().debug(getOverwrittenPropertiesLogMessage(overwrittenProperties, "sysPropsFile"));
+        }
+        overwrittenProperties = result.copyPropertiesFrom(systemPropertyVariables);
+        if (!overwrittenProperties.isEmpty() && getConsoleLogger().isDebugEnabled()) {
+            getConsoleLogger()
+                    .debug(getOverwrittenPropertiesLogMessage(overwrittenProperties, "systemPropertyVariables"));
+        }
+        // We used to take all of our system properties and dump them in with the
+        // user specified properties for SUREFIRE-121, causing SUREFIRE-491.
+        // Not gonna do THAT any more... instead, we only propagate those system properties
+        // that have been explicitly specified by the user via -Dkey=value on the CLI
+        // and only if useUserPropertiesAsSystemProperty is set to true
```

Comment:
I believe that this comment is redundant because wether or not user properties have been passed is at the discretion of the caller. This method shouldn' care about the decision logic.

Reply:
Removed that line in https://github.com/apache/maven-surefire/pull/762/commits/f21ac963663d27d8b316d789648cb6da44088e4d.